.github/workflows/ci-cd.yml

```diff
@@ -17,8 +17,46 @@ env:
   POSTGRES_PASSWORD: postgres
 
 jobs:
+  bundle-install:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+  yarn-install:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Look up if node_module cache exist
+        id: cache
+        uses: actions/cache/restore@v3
+        with:
+          lookup-only: true
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-
+        if: steps.cache.outputs.cache-hit != 'true'
+      - uses: actions/setup-node@v3
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          node-version: 16
+          cache: yarn
+      - run: yarn install --immutable
```

Comment:
`yarn install --immutable` is now the preferred way to check lock file changes on CI.